spot_driver_cpp/src/interfaces/rclcpp_publisher_interface.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <rclcpp/qos.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_publisher_interface.hpp>
+#include <tl_expected/expected.hpp>
+
+namespace {
+constexpr auto kPublisherHistoryDepth = 1;
+
+constexpr auto kImageTopicSuffix = "image";
+constexpr auto kCameraInfoTopicSuffix = "camera_info";
+}  // namespace
+
+namespace spot_ros2 {
+
+RclcppPublisherInterface::RclcppPublisherInterface(const std::shared_ptr<rclcpp::Node>& node) : node_{node} {}
+
+void RclcppPublisherInterface::createPublishers(const std::set<ImageSource>& image_sources) {
+  image_publishers_.clear();
+  info_publishers_.clear();
+
+  for (const auto& image_source : image_sources) {
+    // Since these topic names do not have a leading `/` character, they will be published within the namespace of the
+    // node, which should match the name of the robot. For example, the topic for the front left RGB camera will
+    // ultimately appear as `/MyRobotName/camera/frontleft/image`.
+    const auto topic_name_base = toRosTopic(image_source);
```

Comment:
Is this function getting the namespace of the robot? It doesn't appear to be referencing it below.

Reply:
The namespace of the node is the same as the namespace of the robot, so because the publisher is publishing its topic within the namespace of the node it handles prepending the robot name to the topic name. We don't need to explicitly set the robot namespace within this function as a result of that.